### PR TITLE
Refactor collectors and Bump dependencies+Go

### PR DESCRIPTION
**Pull request by gsanchezgavier**
- Refactors the collection to easily pass more parameters to the process.
- Usual 🧹  updating dependencies , Go 1.18 and removing local snyk

### CODE REVIEW

File: src/metrics.go [hunk 1]
```diff
@@ -16,18 +16,27 @@ import (
...
+	customMetricsQuery  string
+	customMetricsConfig string
+}
+
 // collectMetrics spins off goroutines for each of the metric groups, which
 // send their metrics to the populateMetrics goroutine
-func collectMetrics(db database.DBWrapper, populaterWg *sync.WaitGroup, i *integration.Integration, instanceLookUp map[string]string, customMetricsQuery string, customMetricsConfig string) {
-	defer populaterWg.Done()
+func (mc *metricsCollector) collectMetrics() {
+	defer mc.wg.Done()
```
Comment (alvarocabanas) [inline src/metrics.go:31]: Nit: Since the struct is already called metricsCollecor, I would just call the method collect

File: src/inventory.go [hunk 2]
```diff
@@ -10,10 +10,17 @@ import (
...
+	wg             *sync.WaitGroup
+	instanceLookUp map[string]string
+}
+
 // collectInventory queries the database for the inventory items, then populates
 // the integration with the results
-func collectInventory(db database.DBWrapper, wg *sync.WaitGroup, i *integration.Integration, instanceLookUp map[string]string) {
-	defer wg.Done()
+func (ic *inventoryCollector) collectInventory() {
+	defer ic.wg.Done()
```
Comment (alvarocabanas) [inline src/inventory.go:24]: NIT: same as before I would just call it collect

File: .github/workflows/prerelease.yml [hunk 3]
```diff
@@ -16,21 +16,6 @@ env:
...
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
-          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
-      - name: Scan code for vulnerabilities
-        env:
-          SNYK_TOKEN: ${{ secrets.COREINT_SNYK_TOKEN }}
-        run: make ci/snyk-test
-
```
Comment (alvarocabanas) [inline .github/workflows/prerelease.yml:33]: Are we removing Snyk from the pipelines? Is it that now we run it differently?

**Review by alvarocabanas — APPROVED** [review]
LGTM, just left 2 nits and a question

File: src/metrics.go (hunk 1 above)
Comment (gsanchezgavier) [reply src/metrics.go:31]: that make totally sense :) I will change it thanks

File: .github/workflows/prerelease.yml (hunk 3 above)
Comment (gsanchezgavier) [reply .github/workflows/prerelease.yml:33]: yes since this is running with the SAAS version of snyk

**Pull request merged**